src/blocks/scratch3_procedures.js

```diff
@@ -13,14 +15,34 @@ function Scratch3ProcedureBlocks(runtime) {
 Scratch3ProcedureBlocks.prototype.getPrimitives = function() {
     return {
         'procedures_defnoreturn': this.defNoReturn,
-        'procedures_callnoreturn': this.callNoReturn
+        'procedures_defreturn': this.defReturn,
+        'procedures_callnoreturn': this.callNoReturn,
+        'procedures_callreturn' : this.callReturn,
+        'procedures_report': this.report
+    };
+};
+
+Scratch3ProcedureBlocks.prototype.getHats = function() {
+    return {
+        'procedures_defnoreturn': {
+            restartExistingThreads: false
+        },
+        'procedures_defreturn': {
+            restartExistingThreads: false
```

Comment:
These aren't hat blocks, and the mechanism for starting hats isn't intended for starting procedures.